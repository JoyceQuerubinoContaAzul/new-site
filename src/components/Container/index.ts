export { Container } from './Container';
export type * from './Container.types';
