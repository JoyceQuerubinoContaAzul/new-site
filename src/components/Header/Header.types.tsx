export type HeaderProps = React.ComponentProps<'header'> & {
  noLoginAction?: boolean;
};
