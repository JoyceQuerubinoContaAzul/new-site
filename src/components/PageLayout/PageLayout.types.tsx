export type PageLayoutProps = React.ComponentProps<'main'> & {
  noFooter?: boolean;
};
