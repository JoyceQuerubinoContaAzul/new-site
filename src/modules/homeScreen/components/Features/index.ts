export { default as Feature } from "./Feature";
export { default as CardFeatures } from "./CardFeatures";
export type { FeatureData, BadgeData, FeatureProps, CardFeaturesProps } from "./types"; 