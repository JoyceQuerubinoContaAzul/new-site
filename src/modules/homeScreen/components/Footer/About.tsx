import React from "react";
import Image from "next/image";
const appleStore = "/images/apple-icon.svg";
const googleStore = "/images/google-store-icon.svg";

const footerLinks = [
  { text: "Planos", href: "#" },
  { text: "Imprensa", href: "#" },
  { text: "Trabalhe conosco", href: "#" },
  { text: "Ouvidoria", href: "#" },
  { text: "Termos de Uso", href: "#" },
  { text: "PolÃ­tica de privacidade", href: "#" },
];

const appDownloads = [
  {
    alt: "Download on App Store",
    src: appleStore,
    href: "#",
  },
  {
    alt: "Get it on Google Play",
    src: googleStore,
    href: "#",
  },
];

export const AboutColumn = () => {
  return (
    <div className="flex flex-col">
      <h5 className="mb-4 [font-family:'Ping_Pong-Bold',Helvetica] font-bold text-collection-1-azul-ca-base text-xl tracking-[0] leading-7 whitespace-nowrap">
        Conta Azul
      </h5>

      <ul className="w-fit mt-[-1.00px] [font-family:'Ping_Pong-Regular',Helvetica] font-normal text-collection-1-grayscale-dark text-base tracking-[0] leading-8">
        {footerLinks.map((link, index) => (
          <li key={index}>
            <a
              href={link.href}
              className="hover:underline focus:outline-none focus:underline"
            >
              {link.text}
            </a>
          </li>
        ))}
      </ul>

      <h6 className=" mt-4 [font-family:'Ping_Pong-Bold',Helvetica] font-bold text-collection-1-grayscale-dark text-base tracking-[0] leading-7 whitespace-nowrap">
        Baixe nossos APPS
      </h6>

      <div className="flex flex-col gap-4 mt-4">
        {appDownloads.map((app, index) => (
          <a key={index} href={app.href} className="block">
            <Image
              className="object-cover"
              alt={app.alt}
              src={app.src}
              width={107}
              height={32}
            />
          </a>
        ))}
      </div>
    </div>
  );
};

export default AboutColumn;
