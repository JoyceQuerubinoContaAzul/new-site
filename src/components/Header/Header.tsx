// import Image from 'next/image';
import { cn } from '@/utils/cn';
import { Container } from '@/components/Container';
import type { HeaderProps } from './Header.types';

export const Header = (props: HeaderProps) => {
  const { className, noLoginAction, ...rest } = props;

  return (
    <header
      className={cn(
        'sticky top-0 z-50 bg-white py-3.5 md:static md:pt-[46px] md:pb-7 dark:bg-[#1c191a]',
        className,
      )}
      {...rest}
    >
      <Container className="flex items-center justify-between gap-3">
        <div className="h-[24px] w-[167px]">
          <img src="/images/ca-logo.svg" alt="Logo" />
        </div>

        {!noLoginAction && (
          <div className="md:hidden">
            <p className="text-right text-xs font-medium">
              JÃ¡ tem uma conta?
            </p>
          </div>
        )}
      </Container>
    </header>
  );
};
