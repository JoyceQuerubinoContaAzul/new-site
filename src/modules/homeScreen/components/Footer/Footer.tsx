import React from "react";
import { LegalText } from "./LegalText";
import { Container } from "@/components/Container";
import ListColumnsFooter from "./ListColumnsFooter";

export const Footer = () => {
  return (
    <footer className="mt-32 w-full items-center bg-white">
        <Container className="flex flex-col py-32">
            <ListColumnsFooter/>
            <LegalText />
        </Container>
    </footer>
  );
};

export default Footer;
