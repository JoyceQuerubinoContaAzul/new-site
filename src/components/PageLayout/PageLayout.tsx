import { cn } from '@/utils/cn';
import { Header } from '@/components/Header';
import type { PageLayoutProps } from './PageLayout.types';

export const PageLayout = (props: PageLayoutProps) => {
  const { className, children, noFooter, ...rest } = props;

  return (
    <>
      <Header className="shrink-0" noLoginAction={!!noFooter} />

      <main className={cn(className)} {...rest}>
        {children}
      </main>

      {/* {!noFooter && <Footer className="shrink-0" />} */}
    </>
  );
};
