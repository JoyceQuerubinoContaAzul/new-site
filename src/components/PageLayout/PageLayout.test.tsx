import { render } from '@/tests';
import { Header } from '@/components/Header';
import { Footer } from '@/components/Footer';
import { PageLayout } from './PageLayout';

jest.mock('@/components/Header');
jest.mock('@/components/Footer');

type PageLayoutProps = React.ComponentProps<typeof PageLayout>;

const setup = (props: Partial<PageLayoutProps> = {}) =>
  render(<PageLayout {...props} />);

describe('PageLayout component', () => {
  describe('when the "noFooter" is true', () => {
    it('should pass the value to header "noLoginAction" prop', () => {
      setup({ noFooter: true });

      expect(Header).toHaveBeenCalledWith(
        expect.objectContaining({
          noLoginAction: true,
        }),
        undefined,
      );
    });

    it('should not render the Footer component', () => {
      setup({ noFooter: true });

      expect(Footer).not.toHaveBeenCalled();
    });
  });

  describe('when the "noFooter" is falsy', () => {
    it('should pass the value to header "noLoginAction" prop', () => {
      setup({ noFooter: undefined });

      expect(Header).toHaveBeenCalledWith(
        expect.objectContaining({
          noLoginAction: false,
        }),
        undefined,
      );
    });

    it('should not render the Footer component', () => {
      setup({ noFooter: false });

      expect(Footer).toHaveBeenCalled();
    });
  });

  it('should render the Header', () => {
    setup();

    expect(Header).toHaveBeenCalled();
  });

  it('should render the Footer', () => {
    setup();

    expect(Footer).toHaveBeenCalled();
  });

  it('should render the entire layout', () => {
    const { container } = setup({ children: 'Children' });

    expect(container).toMatchSnapshot();
  });
});
