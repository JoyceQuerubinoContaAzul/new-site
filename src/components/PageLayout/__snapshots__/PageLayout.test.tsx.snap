// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout component should render the entire layout 1`] = `
<div>
  <main
    class="grow py-6 md:py-12"
  >
    <div
      class="mx-auto max-w-[520px] px-5"
    >
      Children
    </div>
  </main>
</div>
`;
